src/main/java/use_case/list/removeMedia/RemoveInteractor.java

```diff
@@ -0,0 +1,119 @@
+package use_case.list.removeMedia;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jetbrains.annotations.Nullable;
+
+import data_access.grade_api.UserRepository;
+import entity.AbstractMedia;
+import entity.MediaCollection;
+import entity.User;
+import use_case.list.ListOutputBoundary;
+import use_case.list.ListOutputData;
+import use_case.list.TableRowDataBuilder;
+
+/**
+ * The list display Interactor.
+ */
+public final class RemoveInteractor implements RemoveInputBoundary {
+    private final UserRepository userRepository;
+    private final ListOutputBoundary listPresenter;
+
+    public RemoveInteractor(UserRepository userRepository, ListOutputBoundary presenter) {
+        this.userRepository = userRepository;
+        this.listPresenter = presenter;
+    }
+
+    /**
+     * Entrypoint for calling the remove interactor to remove a media item from a collection.
+     * @param input The input data containing the collection name and the media name to be removed.
+     */
+    @Override
+    public void execute(RemoveMediaInput input) {
+        final User user = userRepository.get(userRepository.getCurrentUsername());
+        final List<MediaCollection<? extends AbstractMedia>> mediaCollections = user.getAllMediaCollections();
+        final MediaCollection<? extends AbstractMedia> collection = findCollectionWithName(
+                mediaCollections, input.getCollectionName());
+        final AbstractMedia mediaToRemove = findMediaByName(collection, input.getMediaName());
+
+        if (collection == null) {
```

Comment:
Nice job checking for nulls!
Although the order of the checks might need to be rearranged, since when you call `findMediaByName` here, `collection` could be `null`, and it'll throw an error, and that's what the IDE is telling us too via the squigglies.
![image](https://github.com/user-attachments/assets/cb807d38-bb22-44c5-b175-5699e67e2c6d)

What I'd do is instead of calling `findMediaByName` here, just
1. create the variable `mediaToRemove` here without assigning it anything,
2. do a not null check on `collection`, then if true
3. assign `mediaToRemove` via `findMediaByName`, and
4. do a not null check on `mediaToRemove`,  and if that's also true
5. do your success case here (which is the block in your current else statement)
6. otherwise do your null `mediaToRemove` block here
7. and finally do your null `collection` block here
   Although doing it this way would be an ugly nested-if block, so there's probably a better way of doing it.